core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2193,6 +2193,7 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             Optional<MaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, name);
             if (optionalMaterializedView.isPresent()) {
                 MaterializedViewDefinition materializedViewDefinition = optionalMaterializedView.get();
+                analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
```

Comment:
Can you please add a test to `TestAccessControl`? This is a generic change for engine so we should have generic test coverage. I would even consider removing iceberg tests.

Reply:
I will add a test there but I'd prefer to leave an Iceberg test as it exactly covers what was the original problem

Reply:
And perhaps also tests for regular views (if there aren't already), to make sure that they don't have the same issue.

Reply:
consider it done

Reply:
What’s the original problem? Pleas add motivation and details in the PR description.